gemet/thesaurus/views.py

```diff
@@ -201,16 +202,45 @@ def get_context_data(self, **kwargs):
             "concepts": self.concepts,
             "namespace": Term.NAMESPACE,
         })
+        if 'paginator' in kwargs:
+            context.update({
+                "paginator": kwargs['paginator']
+            })
         return context
 
-    def form_valid(self, form):
+    def form_valid(self, form, **kwargs):
         self.query = form.cleaned_data['query']
         self.concepts = search_queryset(
             self.query,
             self.language
         )
+        page = kwargs.get('page', 1)
+        paginator = Paginator(self.concepts, 25)
+        self.concepts = paginator.page(page)
+        return self.render_to_response(
+            self.get_context_data(
+                form=form,
+                paginator=paginator
+            )
+        )
 
-        return self.render_to_response(self.get_context_data(form=form))
+    def get_form_kwargs(self):
+        kwargs = super(SearchView, self).get_form_kwargs()
+        if self.request.method in ('GET', ):
+            kwargs.update({
+                'data': self.request.GET
+            })
+        return kwargs
+
+    def get(self, request, *args, **kwargs):
+        if 'query' in request.GET:
+            form = self.get_form()
+            if form.is_valid():
+                return self.form_valid(form, **kwargs)
+            else:
+                return self.form_invalid(form)
+        else:
+            return super(SearchView, self).get(request, **kwargs)
```

Comment:
All this processing doesn't look right. Maybe inherit from `ListView` and `FormMixin`?